modules/friendlytalkback.js

```diff
@@ -9,13 +9,14 @@
  *** friendlytalkback.js: Talkback module
  ****************************************
  * Mode of invocation:     Tab ("TB")
- * Active on:              Any page with relevant user name (userspace, contribs, etc.)
+ * Active on:              Any page with relevant user name (userspace, contribs, etc.) except IP ranges
  * Config directives in:   FriendlyConfig
  */
 
 Twinkle.talkback = function() {
 
-	if (!mw.config.get('wgRelevantUserName')) {
+	if (!mw.config.exists('wgRelevantUserName') ||
+		mw.util.isIPAddress(mw.config.get('wgRelevantUserName')) !== mw.util.isIPAddress(mw.config.get('wgRelevantUserName'), true)) {
```

Comment:
This is so clever that using it directly is confusing. Also the IP-related functions in Morebits could do with some organisation. How about
```
Morebits.ip = {
	isRange: ... // single line implementation but still worth it
	sanitizeIPv6: ...
	get64Subnet: ...
}
```

?

Reply:
Funny you should mention that... I've got an additional local commit I didn't push up since I wanted to see how the feature was actually implemented.  It adds yet another IP function (we'll likely need to validate the CIDR range isn't too wide) and it proposes just that!  With *two* additional functions, yeah, I absolutely agree.

Reply:
#1297